modules/csv/csv.go

```diff
@@ -30,7 +30,7 @@ func CreateReader(input io.Reader, delimiter rune) *stdcsv.Reader {
 func CreateReaderAndGuessDelimiter(rd io.Reader) (*stdcsv.Reader, error) {
 	var data = make([]byte, 1e4)
 	size, err := rd.Read(data)
-	if err != nil {
+	if err != nil && err != io.EOF {
```

Comment:
So if we get an NPE we fall down to line 37 where we guess a delimiter on an empty []byte and create a reader on the empty input?

Reply:
It actually goes through just fine, but I'll do some testing.

Reply:
Yes, actually, that's what we want. I guess I could do that if the error is io.EOF but I noticed that the data is just an empty byte array and then it just returns comma as the delimiter. It still needs a Reader to show the file on the view page.

Reply:
See my changes to the Create & Guess function